### PR TITLE
AWS KMS keymanager issues (continuation)

### PR DESCRIPTION
**Pull request by marcosy**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**

This PR incorporates the feedback and continues the work done in #2303 (I didn't have permissions to push commits in the same PR)

### CODE REVIEW

File: pkg/server/plugin/keymanager/awskms/awskms.go [hunk 1]
```diff
@@ -854,3 +852,20 @@ func makeFingerprint(pkixData []byte) string {
 	s := sha256.Sum256(pkixData)
 	return hex.EncodeToString(s[:])
 }
+
+// encodeKeyID maps "." and "+" characters to the asciihex value using "_" as
+// escape character. Currently KMS does not support those characters to be used
+// as alias name.
+func encodeKeyID(keyID string) string {
```
Comment (azdagron) [inline pkg/server/plugin/keymanager/awskms/awskms.go:859]: I wonder if we should be more general purpose than this and do proper underscore escaping for any character outside of the allowed set for KMS aliases...

**Review by azdagron — COMMENTED** [review]
Looking pretty good, @marcosy. I had one comment around targeted v.s. general encoding.

File: pkg/server/plugin/keymanager/awskms/awskms.go (hunk 1 above)
Comment (marcosy) [reply pkg/server/plugin/keymanager/awskms/awskms.go:859]: Sounds good, I'll update it

**Comment by marcosy** [comment]
I updated the PR to do an underscore escaping for any character outside of the allowed set for KMS aliases. One thing to take into account is that the current approach asumes that there are not ACME related alias names created in KMS by previous SPIRE versions that include an underscore.

IMO this is a safe assumption because:
- Plugin was not working with ACME and alias creation failed for key ID: `”bundle-acme-acme_account+key"` ('+' char is not supported)
- Key ID for ACME uses the domain name as part of the key ID which generally includes an unsupported character ('.') and does not allows underscores.

**Review by azdagron — APPROVED** [review]
Thanks, @marcosy!

**Pull request merged**